### PR TITLE
Add a few testing guidelines

### PR DESCRIPTION
**Pull request by solonaarmstrong**
Closes: https://github.com/Shopify/web-foundation/issues/64

There are a few preferences that keep coming up in Polaris and I would like to explicitly add them to the style guide.

### CODE REVIEW

File: Best practices/Enzyme.md [hunk 1]
```diff
@@ -8,6 +8,28 @@ We have one utility package for Enzyme, [@shopify/enzyme-utilities](https://gith
...
+  // bad
+  describe('<MyComponent />', () => {
+    it('MyComponent passes the id prop to ChildComponent', () => {});
+  });
+
+  // good
+  describe('<MyComponent />', () => {
+    it('passes the id prop to ChildComponent', () => {});
+  });
+  ```
```
Comment (lemonmade) [inline Best practices/Enzyme.md]: I think these are maybe more appropriate in the general testing guide?

File: Best practices/Enzyme.md [hunk 2]
```diff
@@ -78,3 +100,21 @@ expect(myComponent.find(ChildComponent)).toHaveLength(1);
 // over:
 expect(myComponent.find(ChildComponent).exists()).toBe(true);
 ```
+
+## Event simulations
+
+* When triggering events, trigger the callback function for the event, whenever possible.
```
Comment (lemonmade) [inline Best practices/Enzyme.md]: I think you can combine these into one rule; when triggering props on a composite component, use `trigger`; when doing it for native DOM events, use `simulate`

File: Best practices/React/Testing.md [hunk 3]
```diff
@@ -98,6 +98,18 @@ Another benefit of testing React components is that they typically abstract away
   describe('<MyComponent />', () => {});
   ```
 
+* When there are multiple tests for the same prop or callback function, they should be wrapped in a single describe block
```
Comment (lemonmade) [inline Best practices/React/Testing.md]: Needs punctuation

Comment (lemonmade) [reply Best practices/React/Testing.md]: Also, callback functions are just props, so I might just leave that out.

File: Best practices/React/Testing.md [hunk 4]
```diff
@@ -98,6 +98,18 @@ Another benefit of testing React components is that they typically abstract away
...
+
+  ```js
+  describe('<MyComponent />', () => {
```
Comment (lemonmade) [inline Best practices/React/Testing.md]: Should we should a "bad" example here?

Comment (lemonmade) [reply Best practices/React/Testing.md]: This one is also kind of referenced in https://github.com/Shopify/web-foundation/pull/66/files#diff-eeb8827fd304d15a55965bd445d003e0R9, should we just add the example there?

Comment (solonaarmstrong) [reply Best practices/React/Testing.md]: I think it's a good idea to call this one out explicitly with an example.

File: Best practices/Enzyme.md (hunk 2 above)
Comment (lemonmade) [reply Best practices/Enzyme.md]: You don't need to construct a valid mock event object

File: Best practices/Enzyme.md [hunk 5]
```diff
@@ -78,3 +78,19 @@ expect(myComponent.find(ChildComponent)).toHaveLength(1);
...
+const myComponent = mount(<MyComponent />);
+
+trigger(myComponent, 'onClick');
+```
+
+```js
+const myComponent = mount(<MyComponent />);
+
+myComponent.simulate('click');
+```
```
Comment (lemonmade) [inline Best practices/Enzyme.md]: Needs a newline, might be worth having the counter examples for each of these too.

File: Best practices/React/Testing.md [hunk 6]
```diff
@@ -98,6 +98,48 @@ Another benefit of testing React components is that they typically abstract away
...
+    describe('disabled', () => {
+      it('gets passed to <MySubComponent />', () => {});
+      it('renders if true', () => {});
+      it('does not render if false', () => {});
+    });
+  });
+  ```
+
+* The statement in an it block should use present tense without repeating what is being described.
```
Comment (lemonmade) [inline Best practices/React/Testing.md]: I think this should go in the main testing guide (sorry if I am waffling)

Comment (solonaarmstrong) [reply Best practices/React/Testing.md]: No worries.

**Pull request merged**